### ISSUE TITLE
Multiple Handlers for the same scalar type? 

### ISSUE THREAD

**Issue by HallofFamer** [nikic/scalar_objects#32]
Hello there. I know right now when you assign a new handler, it will simply override the old handler, but I was wondering that in future, is it is possible to allow multiple scalar type handlers, assuming there is no conflict in the methods defined in each handler? Or are there technical difficulties to achieve this in the native C code?

**Comment by nikic** [comment]
I don't think it makes sense to support this natively (it seems too fragile as a default behavior), but you should be able to implement this in userland easily. Just add a handler that implements `__callStatic()` and then dispatch to the appropriate sub-handler that has that method.

**Comment by mikeschinkel** [comment]
@nikic — The biggest problem with not having multiple handlers is when frameworks, libraries or CMSes define their own handler if they do not allow their handler to be overridden.

Maybe you could add the ability to support adding a callback that would be called every time `register_primitive_type_handler()` is called to allow the developer to capture the handler others have added and/or allow the ability to chain handlers?
